emails/models.py

```diff
@@ -673,12 +675,16 @@ def save(self, *args, **kwargs) -> None:
                         break
                     self.address = address_default()
                 locked_profile.update_abuse_metric(address_created=True)
+                locked_profile.last_engagement = datetime.now(timezone.utc)
+                locked_profile.save()
         if not self.user.profile.server_storage:
             self.description = ""
             self.generated_for = ""
             self.used_on = ""
         if not self.user.profile.has_premium:
             self.block_list_emails = False
+        self.user.profile.last_engagement = datetime.now(timezone.utc)
```

Comment:
Do you intend that every change to a random mask causes the user to be marked as engaged?

Reply:
Oh good catch let me clarify that.

Reply:
Removing this because we only want to update the engagement field when the user adds or deletes a mask.